### PR TITLE
Modify logic to determine stretch supervisor

### PR DESCRIPTION
**Pull request by jvrahav**
Modify logic to determine AZ based on the following 1. if there is more than one AZ assume stretched supervisor 2. if there is only one AZ or none assume non-stretched supervisor This logic is arrived at based on the fact that regardless of stretch supervisor or single supervisor, in 8.0+ we always create a zone.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

File: addons/controllers/csi/vspherecsiconfig_utils.go [hunk 1]
```diff
@@ -385,11 +385,9 @@ func (r *VSphereCSIConfigReconciler) isStretchedSupervisorCluster(ctx context.Co
...
 
 	if len(azList.Items) == 1 {
-		// by default every cluster contains legacy availability zone,
-		// if only the legacy AZ is present, return false
-		if azList.Items[0].Name == LegacyZoneName {
-			return false, nil
-		}
+		// by default every non-stretched cluster contains a single zone,
+		// if there is only one AZ we assume it is non-stretched cluster
+		return false, nil
```
Comment (danniel1205) [inline addons/controllers/csi/vspherecsiconfig_utils.go:390]: nit: Would it be a good idea to combine L387 with L383 by using just one IF statement ? That helps prevent `cyclomatic complexity` from exceeding the limit (I know it is not at the moment, but recommend we do our best to simplify the function).

Comment (jvrahav) [reply addons/controllers/csi/vspherecsiconfig_utils.go:387]: makes sense. Handled it ...

**Comment by alfredthenarwhal** [comment]
> @jvrahav: /test

Tests can be triggered only by the trusted reviewers (users mentioned in CODEOWNERS file).

File: addons/controllers/vspherecsiconfig_controller_test.go [hunk 2]
```diff
@@ -274,26 +274,34 @@ var _ = Describe("VSphereCSIConfig Reconciler", func() {
...
 			// zone and set zone to true
-			availabilityZone = &topologyv1alpha1.AvailabilityZone{
+			availabilityZone1 = &topologyv1alpha1.AvailabilityZone{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: "test-az",
 				},
 			}
 
-			Expect(k8sClient.Create(ctx, availabilityZone)).Should(Succeed())
+			availabilityZone1 = &topologyv1alpha1.AvailabilityZone{
```
Comment (maralavi) [inline addons/controllers/vspherecsiconfig_controller_test.go:289]: did you mean ```availabilityZone2``` here instead?

Comment (jvrahav) [reply addons/controllers/vspherecsiconfig_controller_test.go]: i just fixed this in the latest push. can you please check

**Review by maralavi — APPROVED** [review]

**Pull request merged**